com.unity.multiplayer.mlapi/Runtime/Messaging/RpcQueue/RpcQueueProcessor.cs

```diff
@@ -60,7 +62,19 @@ public void ProcessReceiveQueue(NetworkUpdateStage currentStage, bool isTesting)
 
                         if (!isTesting)
                         {
-                            m_NetworkManager.InvokeRpc(currentQueueItem);
+                            try
+                            {
+                                m_NetworkManager.InvokeRpc(currentQueueItem);
+                            }
+                            catch (Exception ex)
+                            {
+                                Debug.LogException(ex);
+
+                                if (NetworkLog.CurrentLogLevel <= LogLevel.Normal)
+                                {
+                                    NetworkLog.LogWarning($"A {currentQueueItem.QueueItemType} threw an exception while executing! Please check Unity logs for more information.");
+                                }
+                            }
```

Comment:
@LukeStampfli — this is what I was thinking
we're logging Unity exception (as if it was an unhandled exception thrown and Unity would log it as an exception).
but also, we're not breaking anything internally within MLAPI, so we just warn dev and suggest looking into it.

Reply:
Yes this looks fine. As long as red appears in the console I'm fine with it 😃